### ISSUE TITLE
a group called "0" cannot be created by provisioning API

### ISSUE THREAD

**Issue by individual-it** [owncloud/core#29698]
### Steps to reproduce
1. try to create a group called "0" with the API e.g. `curl http://admin:admin@$HOST/ocs/v2.php/cloud/groups -d "groupid=0"`

### Expected behaviour
group should be created
```
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>ok</status>
  <statuscode>200</statuscode>
  <message/>
 </meta>
 <data/>
</ocs>
```

### Actual behaviour
cannot create the group
```
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>failure</status>
  <statuscode>400</statuscode>
  <message>Invalid group name</message>
 </meta>
 <data/>
</ocs>
```

### Server configuration
**Operating system**:
Ubuntu 17.10
**Web server:**
Apache
**Database:**
SQLite
**PHP version:**
7.1
**ownCloud version:** (see ownCloud admin page)
10.0.4 RC1 (git)
**Updated from an older ownCloud or fresh install:**
fresh
**Where did you install ownCloud from:**
git

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "oc9xsmdtnlru",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost"
        ],
        "datadirectory": "\/home\/artur\/www\/owncloud-core\/data",
        "overwrite.cli.url": "http:\/\/localhost\/owncloud-core",
        "dbtype": "sqlite3",
        "logtimezone": "UTC",
        "installed": true,
        "default_language": "en",
        "theme": "",
        "loglevel": 0,
        "maintenance": false,
        "log_type": "owncloud",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "debug": false,
        "skeletondirectory": "\/home\/artur\/www\/owncloud-core\/tests\/ui\/skeleton",
        "version": "10.0.4.2",
        "csrf.disabled": true,
        "files_external_allow_create_new_local": "true",
        "singleuser": false,
        "license-key": "***REMOVED SENSITIVE VALUE***"
    }
}
```

**List of activated apps:**

```
Enabled:
  - comments: 0.3.0
  - dav: 0.3.2
  - diagnostics: 0.1.0
  - enterprise_key: 0.1.3
  - federatedfilesharing: 0.3.1
  - federation: 0.1.0
  - files: 1.5.1
  - files_external: 0.7.1
  - files_sharing: 0.10.1
  - files_texteditor: 2.2
  - files_trashbin: 0.9.1
  - files_versions: 1.3.0
  - firewall: 2.4.2
  - gallery: 16.0.2
  - market: 0.2.1
  - notifications: 0.3.1
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - updatenotification: 0.2.1
Disabled:
  - customgroups
  - files_antivirus
  - multidirtest
  - password_policy
  - testing
  - theme-example
  - user_ldap
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
local
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
Firefox
**Operating system:**
Ubuntu 17.10

#### ownCloud log (data/owncloud.log)
```
{"reqId":"MBGMmZ9LUDJKWyXGS7vA","level":3,"time":"2017-11-28T10:13:07+00:00","remoteAddr":"127.0.0.1","user":"admin","app":"provisioning_api","method":"POST","url":"\/owncloud-core\/ocs\/v2.php\/cloud\/groups","message":"Group name not supplied"}
```

**Comment by ownclouders** [comment]
[GitMate.io](https://gitmate.io) thinks the contributor most likely able to help you is @phil-davis.

**Comment by individual-it** [comment]
fixed by  #29712
